clients/lsp-magik.el

```diff
@@ -88,7 +88,8 @@
   :group `lsp-magik
   :package-version '(lsp-mode . "8.0.1"))
 
-(defcustom lsp-magik-java-path (cond ((eq system-type 'windows-nt) "$JAVA_HOME/bin/java")
+(defcustom lsp-magik-java-path (cond ((eq system-type 'windows-nt) (or (executable-find (expand-file-name "bin/java" (getenv "JAVA_HOME")))
```

Comment:
I dont like this to be executed on load time. Imagine every package to perform that on startup similar logic - it will kill lsp-mode's load time.

Reply:
What would you suggest?

Reply:
Use lambda and then lsp-resolve-value

Reply:
Like this?:
```
(defcustom lsp-magik-java-path (cond ((eq system-type 'windows-nt)
                                      (lambda ()
                                        (or (lsp-resolve-value (executable-find (expand-file-name "bin/java" (getenv "JAVA_HOME"))))
                                            (lsp-resolve-value (executable-find "java")))))
                                     (t "java"))
  "Path of the java executable."
  :type 'string
  :group `lsp-magik
  :package-version '(lsp-mode . "8.0.1"))
```

Reply:
to me, it looks like it will be better to wrap the whole thing in a lambda

Reply:
Please review the latest update